src/Filter/ODM/TypeCaster.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder for the canonical source repository
+ * @copyright https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\ApiTools\Doctrine\QueryBuilder\Filter\ODM;
+
+use DateTime;
+use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\TypeCastInterface;
+
+class TypeCaster implements TypeCastInterface
+{
+    public function typeCastField($metadata, $field, $value, $format = null, $doNotTypecastDatetime = false)
+    {
+        if (! isset($metadata->fieldMappings[$field])) {
+            return $value;
+        }
+
+        switch ($metadata->fieldMappings[$field]['type']) {
+            case 'int':
```

Comment:
Do we need `integer` here?

Reply:
Nope, `integer` is not a doctrine ODM mapping type. Not so familiar with Mongo ODM, but looked it up.